code/game/turfs/open/floor/plating.dm

```diff
@@ -152,3 +174,122 @@
 
 /turf/open/floor/plating/foam/tool_act(mob/living/user, obj/item/I, tool_type)
 	return
+
+/turf/open/floor/plating/r_plate
+	name = "reinforced plating"
+	desc = "Thick, tough flooring created with multiple layers of metal."
+	icon_state = "r_plate-0"
+
+	thermal_conductivity = 0.025
+	heat_capacity = INFINITY
+
+	baseturfs = /turf/open/floor/plating
+	allow_replacement = FALSE
+	RCD_proof = TRUE
+	upgradable = FALSE
+
+	var/d_state = PLATE_INTACT
+
+/turf/open/floor/plating/r_plate/examine(mob/user)
+	. += ..()
+	. += deconstruction_hints(user)
+
+/turf/open/floor/plating/r_plate/proc/deconstruction_hints(mob/user)
+	switch(d_state)
+		if(PLATE_INTACT)
+			return span_notice("The plating reinforcements are securely <b>bolted</b> in place.")
+		if(PLATE_BOLTS_LOSENED)
+			return span_notice("The plating reinforcement is <i>unscrewed</i> but <b>welded</b> firmly to the plating.")
+		if(PLATE_CUT)
+			return span_notice("The plating reinforcements have been <i>sliced through</i> but is still <b>loosly</b> held in place.")
+
+/turf/open/floor/plating/r_plate/update_icon_state()
+	icon_state = "r_plate-[d_state]"
+	return ..()
+
+/turf/open/floor/plating/r_plate/attackby(obj/item/W, mob/user, params)
+	user.changeNext_move(CLICK_CD_MELEE)
+	if (!ISADVANCEDTOOLUSER(user))
+		to_chat(user, span_warning("You don't have the dexterity to do this!"))
+		return
+
+	//get the user's location
+	if(!isturf(user.loc))
+		return //can't do this stuff whilst inside objects and such
+
+	add_fingerprint(user)
+
+	var/turf/T = user.loc
+	if(try_decon(W, user, T))
+		return
+	return ..()
+
+/turf/open/floor/plating/r_plate/proc/try_decon(obj/item/W, mob/user, turf/T)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-descriptive-and-obvious-names

For W, T, and for the proc name--`try_deconstruct`